components/filter/filter.js

```diff
@@ -0,0 +1,253 @@
+import '../button/button-icon.js';
+import '../dropdown/dropdown-button-subtle.js';
+import '../dropdown/dropdown-menu.js';
+import '../hierarchical-view/hierarchical-view.js';
+import '../list/list.js';
+import '../list/list-item.js';
+import '../menu/menu.js';
+import '../menu/menu-item.js';
+
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { bodyStandardStyles } from '../typography/styles.js';
+import { getFirstFocusableDescendant } from '../../helpers/focus.js';
+import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+/**
+ * A filter component that contains one or more dimensions a user can filter by.
+ * This component is in charge of all rendering.
+ * @slot - Dimension components used by the filter to construct the different dimensions locally
+ * @fires d2l-filter-change - Dispatched when a dimension's value(s) have changed
+ */
+class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+			 * Disables the dropdown opener for the filter
+			 */
+			disabled: { type: String, reflect: true },
+			_activeDimensionKey: { type: String },
+			_dimensions : { type: Array }
```

Comment:
I'm not totally sure what the default is here, but you may want `attribute: false` for these just to make sure people can't go `<d2l-filter _dimensions="blah">` and mess stuff up.